DTS-STN/passport-status
# Modal with dialog

## [ADO-1888](https://dev.azure.com/DTS-STN/passport-status/_workitems/edit/1888)

### Description
This PR changes the Modal component to use dialog so that it is properly announced by the screen reader immediately as it pops up. This PR also adds an event listener that allows the user to close out of the modal using the ESC button.

### What to test for/How to test
1. Pull in branch
2. Type `npm run dev`
3. Fire up your favorite screen reader (NVDA, JAWS, Orca, etc.)
4. Navigate to `/status`
5. Clicking the `Cancel` button should first announce the page title (Check the status of your passport application...) and then the content of the modal.
6. Ensure that you can close the modal by both clicking `No` and by pressing ESC
